### PR TITLE
Clarify that servers must reject disallowed revocation reasonCodes

### PR DESCRIPTION
**Pull request by rolandshoemaker**
Instead of leaving it ambiguous.

### CODE REVIEW

File: draft-ietf-acme-acme.md [hunk 1]
```diff
@@ -1751,7 +1751,8 @@ reason (optional, int):
 to be used when generating OCSP responses and CRLs. If this field is not set
 the server SHOULD use the unspecified (0) reasonCode value when generating OCSP
 responses and CRLs. The server MAY disallow a subset of reasonCodes from being
-used by the user.
+used by the user. If a request contains a disallowed reasonCode the server MUST
+it.
```
Comment (jsha) [inline draft-ietf-acme-acme.md]: missing "reject"

File: draft-ietf-acme-acme.md [hunk 2]
```diff
@@ -1751,7 +1751,8 @@ reason (optional, int):
 to be used when generating OCSP responses and CRLs. If this field is not set
 the server SHOULD use the unspecified (0) reasonCode value when generating OCSP
 responses and CRLs. The server MAY disallow a subset of reasonCodes from being
-used by the user.
+used by the user. If a request contains a disallowed reasonCode the server MUST
+reject it.
```
Comment (jsoref) [inline draft-ietf-acme-acme.md]: I'd request that there be a defined Error code for this case.

File: draft-ietf-acme-acme.md [hunk 3]
```diff
@@ -1752,7 +1753,9 @@ to be used when generating OCSP responses and CRLs. If this field is not set
 the server SHOULD use the unspecified (0) reasonCode value when generating OCSP
 responses and CRLs. The server MAY disallow a subset of reasonCodes from being
 used by the user. If a request contains a disallowed reasonCode the server MUST
-reject it.
+reject it with the error type "urn:ietf:params:acme:error:badRevocationReason".
+The problem document returned with the error MUST include an "supportedReasons"
```
Comment (jsoref) [inline draft-ietf-acme-acme.md]: an => a

File: draft-ietf-acme-acme.md [hunk 4]
```diff
@@ -1751,7 +1752,10 @@ reason (optional, int):
 to be used when generating OCSP responses and CRLs. If this field is not set
 the server SHOULD use the unspecified (0) reasonCode value when generating OCSP
 responses and CRLs. The server MAY disallow a subset of reasonCodes from being
-used by the user.
+used by the user. If a request contains a disallowed reasonCode the server MUST
+reject it with the error type "urn:ietf:params:acme:error:badRevocationReason".
+The problem document returned with the error MUST include a "supportedReasons"
```
Comment (jsha) [inline draft-ietf-acme-acme.md]: I'm reluctant to add extra infrastructure (the "supportedReasons" field) for a use case that will be exceedingly rare. We don't really expect most clients to implement some sort of fancy menu like "Try one of these other revocation reasons!" Instead how about we just say "the problem detail SHOULD indicate which revocation reasons are allowed."

**Pull request merged**